### ISSUE TITLE
Empty result with arki-query --sort

### ISSUE THREAD

**Issue by edigiacomo** [ARPA-SIMC/arkimet#212]
```
$ arki-query --debug --data --verbose --sort=day:timerange 'Reftime:=2020-02-13 00  ; Product : GRIB1,80,002,033 or GRIB1,80,002,34 or GRIB1,80,201,187  ; Level : GRIB1,105,10'  http://arkimet.metarpa:8090/dataset/cosmo_5M_ita
> /tmp/buttami.grib
$ grib_count /tmp/buttami.grib
0
```

Without `--sort`, everything works fine

```
$ arki-query --debug --data --verbose 'Reftime:=2020-02-13 00  ; Product : GRIB1,80,002,033 or GRIB1,80,002,34 or GRIB1,80,201,187  ; Level : GRIB1,105,10'  http://arkimet.metarpa:8090/dataset/cosmo_5M_ita > /tmp/buttami.grib >
 /tmp/buttami.grib
$ grib_count /tmp/buttami.grib
219
```

I have replicated the issue with arkimet client v1.17 and v1.20: i think it's an arki-server problem, because it started to occur after the arki-server update to v1.20.

Unfortunately, `--verbose --debug` options of the client don't produce any log.

Below the arki-server log (`/var/log/arkimet/server-error.log`):

```
2020-02-13 07:24:27,398 ERROR Exception caught after headers have been sent
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/arkimet/server/views.py", line 281, in run
    self.stream()
  File "/usr/lib/python3.6/site-packages/arkimet/server/views.py", line 327, in stream
    data_start_hook=self.send_headers)
RuntimeError: cannot retrieve data: BLOB source has no reader associated
```

**Comment by spanezz** [comment]
Looking into it now

**Comment by spanezz** [comment]
Reproduced in test suite

**Comment by spanezz** [comment]
Fixed in 96f04b19692ec3508583fd096e606c6427fb16b7, where I mistyped the issue number in the commit message :-/
